src/Reflection/ReflectionClass.php

```diff
@@ -319,35 +324,12 @@ public function inNamespace() : bool
      */
     private function scanMethods() : array
     {
-        // merging together methods from interfaces, parent class, traits, current class (in this precise order)
-        /** @var \ReflectionMethod[] $inheritedMethods */
-        $inheritedMethods = \array_merge(
-            \array_merge(
-                [],
-                ...\array_map(
-                    function (ReflectionClass $ancestor) : array {
-                        return $ancestor->getMethods();
-                    },
-                    \array_values(\array_merge(
-                        $this->getInterfaces(),
-                        \array_filter([$this->getParentClass()])
-                    ))
-                ),
-                ...\array_map(
-                    function (ReflectionClass $trait) : array {
-                        return \array_map(function (ReflectionMethod $method) use ($trait) : ReflectionMethod {
-                            return ReflectionMethod::createFromNode(
-                                $this->reflector,
-                                $method->getAst(),
-                                $this->declaringNamespace,
-                                $trait,
-                                $this
-                            );
-                        }, $trait->getMethods());
-                    },
-                    $this->getTraits()
-                )
-            ),
+        // Note: methods are not merged via their name as array index, since internal PHP method
```

Comment:
Split to a private method, put the comment on that private method

Reply:
Is it necessary? The comment looks same as here: https://github.com/Roave/BetterReflection/commit/ca631dd343a940e9fc939c566ba02051dd311898

And I think there's already enough private methods :) `getMethods()` calls private method `getMethodsIndexedByName()` and this method calls private method `scanMethods()` :)

Reply:
@kukulich yes - the number of private methods is not scary, but their singular size is. The fact that there is a comment here should move the comment to the method, rather than inline

Reply:
I've tried to improve it in special commit: https://github.com/Roave/BetterReflection/pull/350/commits/b9046e81537a13613cb484c6d1660a33067ab7f4